Install TIRPC headers in CI to fix asyn build.

gustavosr8 opened the pull request:
GitHub Actions runners lack libtirpc-dev by default, causing missing rpc.h errors during ASYN dependency preparation. Install libtirpc-dev unconditionally for all Linux jobs to satisfy ASYN's default TIRPC=YES configuration [1].

[1] https://github.com/epics-modules/asyn/blame/d55786e0508b1f8244cfae943ebc5fffccfb7590/configure/CONFIG_SITE.Common.linux-x86_64#L8

gustavosr8 committed:
Install TIRPC headers in CI to fix asyn build.

GitHub Actions runners lack libtirpc-dev by default, causing missing
rpc.h errors during ASYN dependency preparation. Install libtirpc-dev
unconditionally for all Linux jobs to satisfy ASYN's default TIRPC=YES
configuration [1].

[1] https://github.com/epics-modules/asyn/commit/0ed37b298307f3c32718bd6b28fcdf4bd459bacf

gustavosr8 committed:
Update deprecated runner image.

Ubuntu-20.04 image is deprecated [1], causing the check to fail. Switch
to ubuntu-22.04 to avoid the deprecation.

[1] https://github.com/actions/runner-images/issues/11101

gustavosr8 commented:
@ericonr @henriquesimoes wdyt?

gustavosr8 commented:
Just notice something... it still need to run [asyn-config.sh](https://github.com/areaDetector/ADCore/blob/master/.ci-local/asyn-config.sh) since https://github.com/epics-modules/asyn/blame/d55786e0508b1f8244cfae943ebc5fffccfb7590/configure/CONFIG_SITE.Common.linux-x86_64#L8 ?

henriquesimoes commented on .github/workflows/ci-scripts.yml:
```diff
@@ -49,7 +49,7 @@ jobs:
             deps: adsupport
 
-          - os: ubuntu-20.04
+          - os: ubuntu-latest
```
As we discussed in person, maybe we should simply bump to 22.04, so it is still tested with older-ish software.

Moving to a container-based deploy would be the best option in the long run, but probably also harder to implement now.

henriquesimoes reviewed (commented):
I think you linked the wrong commit in the commit message. It should be https://github.com/epics-modules/asyn/commit/0ed37b298307f3c32718bd6b28fcdf4bd459bacf. I usually prefer linking the commit itself when possible instead of the blame page.

gustavosr8 committed:
Remove the asyn configuration hook.

asyn-config.sh was being used in ASYN_HOOK in order to set TIRPC=YES
when libtirpc headers were found. We don't need to set this anymore
as TIRPC=YES is now the default configure for asyn builds on linux
[1].

[1] https://github.com/epics-modules/asyn/commit/0ed37b298307f3c32718bd6b28fcdf4bd459bacf

henriquesimoes reviewed (approved):
Looks fine by me.
